alphagov/di-txma-audit
# Tma-431 new reingest metrics

For the re-ingest functionality, we introduce some cloudwatch metrics and alarms for when events go through the re-ingest stage. 

# Features

for the buckets with `-fail` in their name, we have metrics for if 1 or more events appear in 24 hour period. The events we monitor are

- PutRequest
- DeleteRequest

# ⚠️ Breaking Changes

Each lambda has an associated Cloudwatch Log group, the log group name follows the format `/aws/lambda/<FUNCTION_NAME>`. For the function `ReIngestFunction`, its log group was called `/aws/lambda/ReIngest`, this meant that AWS had to automatically create a log group called `/aws/lambda/ReIngestFunction` which does not have the necessary properties (KMS encryption and retention period).

While the fix is simple, it does require some manual intervention of deleting old log groups which will need to happen across ALL environments.

## Steps

1. Delete the log groups `/aws/lambda/ReIngest` and `/aws/lambda/ReIngestFunction`
2. Deploy CloudFormation, it should now re-create the log groups
3. To test, 
    1. go to the Reingest Lambda on the AWS console
    2. Select the **Monitor** tab and select **View Logs in Cloudwatch**
    3. the newly created log group should have a creation time of **x minutes ago**
